src/Pagination.js

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import BootstrapMixin from './BootstrapMixin';
+
+const Pagination = React.createClass({
+  mixins: [BootstrapMixin],
+
+  propTypes: {
+    activePage: React.PropTypes.number,
+    items: React.PropTypes.number,
+    maxButtons: React.PropTypes.number,
+    ellipsis: React.PropTypes.bool,
+    first: React.PropTypes.bool,
+    last: React.PropTypes.bool,
+    prev: React.PropTypes.bool,
+    next: React.PropTypes.bool,
+    onSelect: React.PropTypes.func
+  },
+
+  getDefaultProps() {
+    return {
+      activePage: 1,
+      items: 1,
+      maxButtons: null,
+      first: false,
+      last: false,
+      prev: false,
+      next: false,
+      ellipsis: true,
+      bsClass: 'pagination'
+    };
+  },
+
+  handleSelect(pagenumber) {
+    if(this.props.onSelect){
+      this.props.onSelect(pagenumber);
+    }
+  },
+
+  renderButton({child, key, className, onClick}){
+    //TODO: replace <a> with <SafeAnchor> when it is avaliable
+    return (
+      <li
+        key={key}
+        onClick={onClick}
+        className={className}>
+        <a href='#' onClick={e => e.preventDefault()}>{child}</a>
+      </li>
+    );
+  },
+
+  renderPageButtons() {
+    let pageButtons = [];
+    let startPage, endPage, hasHiddenPagesBefore, hasHiddenPagesAfter;
+
+    if(this.props.maxButtons){
+      hasHiddenPagesBefore = this.props.activePage - 1 > 1;
+      startPage = hasHiddenPagesBefore ? this.props.activePage - 1 : 1;
+      hasHiddenPagesAfter = startPage + this.props.maxButtons <= this.props.items;
+
+      if(!hasHiddenPagesAfter){
+        endPage = this.props.items;
+        startPage = this.props.items - this.props.maxButtons + 1;
+      } else {
+        endPage = startPage + this.props.maxButtons - 1;
+      }
+    } else {
+      startPage = 1;
+      endPage = this.props.items;
+    }
+
+    for(let pagenumber = startPage; pagenumber <= endPage; pagenumber++){
+      pageButtons.push(
+        this.renderButton({
+          child: pagenumber,
+          key: pagenumber,
+          className: pagenumber === this.props.activePage ? 'active' : null,
+          onClick: this.handleSelect.bind(this, pagenumber)
+        })
+      );
+    }
+
+    if(this.props.maxButtons && hasHiddenPagesAfter && this.props.ellipsis){
+      pageButtons.push(
+        this.renderButton({
+          child: (<span aria-label='More'>...</span>),
+          key: 'ellipsis',
+          className: 'disabled'
+        })
+      );
+    }
+
+    return pageButtons;
+  },
+
+  renderPrev() {
+    return this.props.prev ?
+      this.renderButton({
+        child: (<span aria-label='Previous'>&lsaquo;</span>),
+        key: 'prev',
+        className: this.props.activePage === 1 ? 'disabled' : null,
+        onClick: this.handleSelect.bind(this, this.props.activePage - 1)
+      }) : null;
```

Comment:
Instead of ternaries can you just use gate clauses. It's slightly easier on the eyes to read.